lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -54,13 +57,33 @@ def timeout_message
     end
 
     def deploy_timed_out?
-      super || @latest_rs && @latest_rs.deploy_timed_out?
+      if @progress
+        @progress["status"] == 'False'
+      else
+        super || @latest_rs && @latest_rs.deploy_timed_out?
+      end
     end
 
     def exists?
       @found
     end
 
+    def timeout
+      raw_json, _err, st = kubectl.run("get", type, @name, "--output=json")
```

Comment:
Since there isn't a default value for this (i.e. if it isn't set in the template, it won't be set ever), we can look at `@definition["spec"]["progressDeadlineSeconds"]` for this instead of querying the API. As a rule, I try to keep all API queries inside the `sync` method. This reduces the number of overall calls (since e.g. `deploy_succeeded?` might get called a bunch of times in a row), and also makes sure we keep a consistent view of the cluster within a given polling cycle.

EDIT: In the end I don't think we should override this anyway (see my other comment).